apps/test/unit/p5lab/spritelab/actionCommandsTest.js

```diff
@@ -451,4 +452,226 @@ describe('Action Commands', () => {
       ])
     ).to.equal(-90);
   });
+
+  // tests copied from dance party repo here:
+  // https://github.com/code-dot-org/dance-party/blob/main/test/unit/layoutTest.js
+  describe('layoutSprites', () => {
+    const minX = 20;
+    const maxX = 400 - minX;
+    const minY = 35;
+    const maxY = 400 - 40;
+
+    const createSprites = (n, animation) => {
+      for (let i = 0; i < n; i++) {
+        coreLibrary.addSprite({
+          animation: animation
+        });
+      }
+    };
+
+    beforeEach(() => {
+      let image = new p5.Image(100, 100, coreLibrary.p5);
+      let frames = [{name: 0, frame: {x: 0, y: 0, width: 50, height: 50}}];
+      let sheet = new coreLibrary.p5.SpriteSheet(image, frames);
+      let animation = new coreLibrary.p5.Animation(sheet);
+      coreLibrary.p5._predefinedSpriteAnimations = {
+        cat: animation,
+        alien: animation,
+        duck: animation
+      };
+    });
+
+    it('circle layout works with 1 sprite', () => {
+      coreLibrary.addSprite({
+        animation: 'cat'
+      });
+
+      commands.layoutSprites.apply(coreLibrary, ['cat', 'circle']);
+      const sprites = coreLibrary.getSpriteArray({costume: 'cat'});
+
+      // one sprite, facing upwards
+      expect(sprites.length).to.equal(1);
+      expect(sprites[0].x).to.equal(200);
+      expect(sprites[0].rotation).to.equal(0);
+    });
+
+    it('circle layout works with 2 sprites', () => {
+      createSprites(2, 'cat');
+
+      commands.layoutSprites.apply(coreLibrary, ['cat', 'circle']);
+      const sprites = coreLibrary.getSpriteArray({costume: 'cat'});
+
+      expect(sprites.length).to.equal(2);
+      expect(sprites[0].x).to.equal(200);
+      expect(sprites[0].rotation).to.equal(0);
+
+      expect(sprites[1].x).to.equal(200);
+      expect(sprites[1].rotation).to.equal(180);
+
+      // fairly weak test that they just have different y values
+      expect(sprites[0].y).to.not.equal(sprites[1].y);
+    });
+
+    it('circle changes radius/scale as we add more sprites', () => {
+      createSprites(2, 'cat');
+      createSprites(10, 'alien');
+      createSprites(20, 'duck');
+
+      commands.layoutSprites.apply(coreLibrary, ['cat', 'circle']);
+      const cats = coreLibrary.getSpriteArray({costume: 'cat'});
+
+      commands.layoutSprites.apply(coreLibrary, ['alien', 'circle']);
+      const aliens = coreLibrary.getSpriteArray({costume: 'alien'});
+
+      commands.layoutSprites.apply(coreLibrary, ['duck', 'circle']);
+      const ducks = coreLibrary.getSpriteArray({costume: 'duck'});
+
+      // fewer cats, so they should be bigger
+      expect(cats[0].scale).to.be.greaterThan(aliens[0].scale);
```

Comment:
It looks like we do test that the sprite re-sizing works as expected. Should we consider adding a simple check to one of the other layouts to make sure that the sprites get resized for it as well? Again, following the Dance logic, I think we should set each sprite's scale to `30` for all layouts besides "circle".

Reply:
Yes! Would have caught the sizing oversight, will add

Reply:
Note that we'll probably want to move to `getScale()` here as well as throughout this file. It's possible the tests will still pass because the test sprites aren't using actual animations with different dimensions.